Doc and release note for manifest options_page key support

rebloor opened the pull request:
### Description

Added details of aliasing of options_ui.page to options_page and a release note.

#### Related issues

Addresses the content requirements of [Bug 1816960](https://bugzilla.mozilla.org/show_bug.cgi?id=1816960) Consider aliasing options_page to options_ui.page

BCD updates in https://github.com/mdn/browser-compat-data/pull/22877.

rebloor committed:
Doc and release note for manifest options_page key support

rpl commented on files/en-us/mozilla/add-ons/webextensions/manifest.json/options_page/index.md:
```diff
@@ -32,15 +32,13 @@ browser-compat: webextensions.manifest.options_page
+The way the user opens the page is browser-dependent. In Firefox, the page opens when the extension's icon is clicked. Your extension can also open the page using {{WebExtAPIRef("runtime.openOptionsPage()")}}.
 
-Use the `options_page` key to define an [options page](/en-US/docs/Mozilla/Add-ons/WebExtensions/user_interface/Options_pages) for your extension.
+Alternatively, you can use the [`options_ui`](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/options_ui) key with `open_in_tab` set to `true`. In Firefox, if `options_ui` is specified, `options page` is ignored.
```
I was wondering if this was really a Firefox specific behavior and so I gave it a try on Chrome and Chrome seems to also be ignoring option_page if options_ui is specified (I haven't tried on Safari though, and so I'm not sure if that's the case on Safari as well, but it is not specific to Firefox).

rebloor replied:
```suggestion
Alternatively, you can use the [`options_ui`](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/options_ui) key with `open_in_tab` set to `true`. If `options_ui` is specified, `options page` is ignored.
```

rpl reviewed (approved):
lgtm, @rebloor the only comment below is related to the part where in options_page we state that ignoring option_page when options_ui is also included in the manifest is Firefox specific, that seems to be the case in Chrome as well (and so we may remove the "In Firefox" part of it, if turns out Safari diverged on this behavior we can consider to change that part later).

rebloor committed:
Feedback update

Removed reference to Firefox regarding `options page` ignored when`options_ui` is specified. Seems to be the same in Chrome

rebloor committed:
Merge branch 'main' into manifest-options_page-support-doc-and-rn
